src/equations/equations.jl

```diff
@@ -48,48 +69,122 @@ dispatching on the return type.
+this simply involves a conversion from  primitive to conservative variables. The intention 
+here is to have a set of test cases (for example, [`initial_condition_gaussian`](@ref)) for 
+which the initial condition is prescribed using a standardized set of primitive variables 
+in a global Cartesian coordinates, and transformed to the specific prognostic variables 
+required for a given model.
+!!! note 
+    When using the covariant formulation, the initial velocity components should be defined 
+    in the coordinate system specified by the `GlobalCoordinateSystem` type parameter in
+    [`AbstractCovariantEquations`](@ref).
+"""
+function transform_initial_condition(initial_condition, ::AbstractCovariantEquations)
+    function initial_condition_transformed(x, t, aux_vars, equations)
+        return Trixi.prim2cons(global2contravariant(initial_condition(x, t, equations),
+                                                    aux_vars, equations), aux_vars,
+                               equations)
+    end
+    return initial_condition_transformed
+end
+
+# Version for standard Cartesian formulations
+function transform_initial_condition(initial_condition, ::AbstractEquations)
+    function initial_condition_transformed(x, t, equations)
+        return Trixi.prim2cons(initial_condition(x, t, equations), equations)
+    end
+    return initial_condition_transformed
+end
+
+"""
+    contravariant2global(u, aux_vars, equations)
+
+Transform the vector `u` of solution variables with the momentum or velocity given in terms 
+of local contravariant components into the global coordinate system specified by the 
+`GlobalCoordinateSystem` type parameter in [`AbstractCovariantEquations`](@ref). `u` is a 
+vector type of the correct length `nvariables(equations)`. Notice the function doesn't 
+include any error checks for the purpose of efficiency, so please make sure your input is 
+correct. The inverse conversion is performed by [`global2contravariant`](@ref).
+"""
+function contravariant2global end
+
+"""
+    global2contravariant(u, aux_vars, equations)
+
+Transform the vector `u` of solution variables with momentum or velocity components
+given with respect to the global coordinate system into local contravariant components. The 
+global coordinate system is specified by the `GlobalCoordinateSystem` type parameter in 
+[`AbstractCovariantEquations`](@ref). `u` is a vector type of the correct length 
+`nvariables(equations)`. Notice the function doesn't include any error checks for the 
+purpose of efficiency, so please make sure your input is correct. The inverse conversion is 
+performed by [`contravariant2global`](@ref).
+"""
+function global2contravariant end
 
 # If no auxiliary variables are passed into the conversion to spherical coordinates, do not 
 # do any conversion.
-@inline contravariant2spherical(u, equations) = u
+@inline contravariant2global(u, ::AbstractEquations) = u
```

Comment:
Not needed anymore?
```suggestion
```

Reply:
Yes, this is not needed. I will delete and then merge.